rustler/src/resource.rs

```diff
@@ -145,9 +144,28 @@ where
         }
     }
 
+    pub fn make_binary<'env, F>(&self, env: Env<'env>, f: F) -> Binary<'env>
+    where
+        F: FnOnce(&T) -> &[u8],
```

Comment:
Since the lifetime of the `&T` and `&[u8]` is important for soundness and them being accidentally different would not be a compile error here, maybe we should annotate it explicitly to make that more clear?

`F: FnOnce(&'a T) -> &'a [u8]`

We might also want to note the importance of that contract as a comment within the implementation, above the unsafe block?

Reply:
I agree, will do.